lib/src/source/git.dart

```diff
@@ -426,6 +433,22 @@ class BoundGitSource extends CachedSource {
     _updatedRepos.add(path);
   }
 
+  /// Check if [directory] is a valid git repo. If invalid and
+  /// [delIfInvalid] is set to be true, it will be deleted.
+  ///
+  bool _checkIfValidGitRepo(Directory directory, {bool delIfInvalid = false}) {
+    if (directory.existsSync()) {
+      var result = runProcessSync(git.command, ['log', '-1'],
+          workingDir: directory.path);
+      if (result.exitCode == 0) {
+        return true;
+      } else if (delIfInvalid == true) {
+        directory.deleteSync(recursive: true);
```

Comment:
Let's use `deleteEntry` from `lib/src/io.dart`, I assume it exists in there for a reason..